### PR TITLE
Add roles for LV

### PR DESCRIPTION
**Pull request by ashmitha7**

### CODE REVIEW

File: roles/system/glusterfs_lv/lvchange/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+lvchange
+========
+
+lvchange changes certain parameters of an existing LogicalVolume
+
+Requirements
+------------
+lvrename allows you to change certain parameters of a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvchange/README.md]: "The volume groups on which the logical volume will be created needs to exist."

File: roles/system/glusterfs_lv/lvchange/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Role Variables
+--------------
+
+| Parameters      | Required | Default | Choices        | Description                                         |
+| ----------      | -------- | ------- | -------        | -----------                                         |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvchange/README.md]: 'state' variable description is missing.

File: roles/system/glusterfs_lv/lvchange/tests/vars.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+# vars file for lvchange
+lvchange_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+lv1:
+  state: change
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvchange/tests/vars.yml]: 'state' var needn't be specific to each lv.
state: change - will be same for all lvs, hence can be declared just once.

File: roles/system/glusterfs_lv/lvcreate/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+lvcreate
+=========
+
+lvcreate role allows to create a LogicalVolume
+
+Requirements
+------------
+lvcreate allows you to create a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvcreate/README.md]: change the requirements, same as lvchange.

File: roles/system/glusterfs_lv/lvcreate/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+Role Variables
+-------------
+
+| Parameters      | Required | Default | Choices                                      | Description                                                     |
+| ----------      | -------- | ------- | -------                                      | -----------                                                     |
+|cachemode        |no        |         |**passthrough**/**writeback**/**writethrough**|Determines when the writes to a cache LV are considered complete |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvcreate/README.md]: missing 'state' variable.

File: roles/system/glusterfs_lv/lvcreate/tests/vars.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+---
+# vars file for lvcreate
+lvcreate_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+lv1:
+  state: create
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvcreate/tests/vars.yml]: 'state' variable will be same for all lvs.

File: roles/system/glusterfs_lv/lvextend/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+lvextend
+=========
+
+lvextend role allows to extend a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvextend/README.md]: nit: s/allows/allows you

File: roles/system/glusterfs_lv/lvextend/README.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+Requirements
+------------
+lvextend allows you to extend a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvextend/README.md]: change requirements, same as lvchange.

File: roles/system/glusterfs_lv/lvextend/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+Role Variables
+--------------
+
+| Parameters      | Required | Default | Choices        | Description                                |
+| ----------      | -------- | ------- | -------        | -----------                                |
+|extents          |no        |         |                |Change/set the lv size in units of extents  |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvextend/README.md]: missing 'state' variable.

File: roles/system/glusterfs_lv/lvextend/tests/vars.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
+---
+# vars file for lvextend
+lvextend_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+lv1:
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvextend/tests/vars.yml]: 'state' variable will be common for all lvs.

File: roles/system/glusterfs_lv/lvreduce/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+lvreduce
+=========
+
+lvreduce role allows to reduce a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvreduce/README.md]: nit: "lvreduce allows you to reduce a LogicalVolume"

File: roles/system/glusterfs_lv/lvreduce/README.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+Requirements
+------------
+lvreduce allows you to reduce a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvreduce/README.md]: change requirements as suggested above.

File: roles/system/glusterfs_lv/lvreduce/README.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+Role Variables
+--------------
+
+| Parameters      | Required | Default | Choices        | Description                                |
+| ----------      | -------- | ------- | -------        | -----------                                |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvreduce/README.md]: missing 'state' variable.

File: roles/system/glusterfs_lv/lvreduce/tests/vars.yml [hunk 14]
```diff
@@ -0,0 +1,23 @@
+---
+# vars file for lvreduce
+lvreduce_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+lv1:
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvreduce/tests/vars.yml]: common 'state' variable, as explained above.

File: roles/system/glusterfs_lv/lvremove/README.md [hunk 15]
```diff
@@ -0,0 +1,30 @@
+lvremove
+=========
+
+lvremove role allows to remove a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvremove/README.md]: nit: "lvremove allows you to remove a LogicalVolume"

File: roles/system/glusterfs_lv/lvremove/README.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+
+Requirements
+------------
+lvremove allows you to remove a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvremove/README.md]: change requirements as suggested above.

File: roles/system/glusterfs_lv/lvremove/README.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+
+Role Variables
+--------------
+
+| Parameters      | Required | Default | Choices        | Description                                |
+| ----------      | -------- | ------- | -------        | -----------                                |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvremove/README.md]: missing 'state', 'vgname', 'lvname' variables.

File: roles/system/glusterfs_lv/lvremove/tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,8 @@
+---
+# tasks file for lvremove
+- name: Remove logical volumes
+  lv: state={{state}}
+      vgname={{item.vgname}}
+      lvname={{item.lvname}}
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvremove/tasks/main.yml]: you explained a 'force' variable in the README, missing 'force' variable here.

File: roles/system/glusterfs_lv/lvremove/tests/vars.yml [hunk 19]
```diff
@@ -0,0 +1,23 @@
+---
+# vars file for lvremove
+lvremove_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+- "{{ tp1 }}"
+- "{{ tp2 }}"
+lv1:
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvremove/tests/vars.yml]: common 'state' variable as suggested above.

File: roles/system/glusterfs_lv/lvremove/tests/vars.yml [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+  state: remove
+  vgname: vg1
+  lvname: lvol1
+lv2:
+  state: remove
+  vgname: vg2
+  lvname: lvol1
+tp1:
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvremove/tests/vars.yml]: tp1 and tp2 would be?

File: roles/system/glusterfs_lv/lvrename/README.md [hunk 21]
```diff
@@ -0,0 +1,32 @@
+lvrename
+=========
+
+lvrename renames an existing LogicalVolume
+
+Requirements
+------------
+lvrename allows you to rename a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvrename/README.md]: change the requirements as suggested above.

File: roles/system/glusterfs_lv/lvrename/README.md [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+
+Role Variables
+--------------
+
+| Parameters      | Required | Default | Choices        | Description                  |
+| ----------      | -------- | ------- | -------        | -----------                  |
+|lvname           |no        |         |**yes**         | Name of the LogicalVolume    |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvrename/README.md]: missing 'state' variable.

File: roles/system/glusterfs_lv/lvrename/tests/vars.yml [hunk 23]
```diff
@@ -0,0 +1,15 @@
+---
+# vars file for lvrename
+lvrename_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+lv1:
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvrename/tests/vars.yml]: common 'state' variable, as suggested above.

File: roles/system/glusterfs_lv/lvresize/README.md [hunk 24]
```diff
@@ -0,0 +1,35 @@
+lvresize
+=========
+
+lvresize role allows to resize a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvresize/README.md]: nit: "lvresize allows you to resize a LogicalVolume."

File: roles/system/glusterfs_lv/lvresize/README.md [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+
+Requirements
+------------
+lvresize allows you to resize a LogicalVolume
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvresize/README.md]: change requirements as suggested above.

File: roles/system/glusterfs_lv/lvresize/README.md [hunk 26]
```diff
@@ -0,0 +1,35 @@
...
+
+Role Variables
+--------------
+
+| Parameters      | Required | Default | Choices        | Description                                |
+| ----------      | -------- | ------- | -------        | -----------                                |
+|extents          |no        |         |                |Change/set the lv size in units of extents  |
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvresize/README.md]: missing 'state' variable.

File: roles/system/glusterfs_lv/lvresize/tests/vars.yml [hunk 27]
```diff
@@ -0,0 +1,22 @@
+---
+# vars file for lvresize
+lvresize_list:
+- "{{ lv1 }}"
+- "{{ lv2 }}"
+
+lv1:
+  state: resize
```
Comment (devyanikota) [inline roles/system/glusterfs_lv/lvresize/tests/vars.yml]: common 'state' variable, as suggested above.

**Comment by sac** [comment]
This is moved to gluster.infra.

**Pull request closed without merging**